Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/EventLogServiceImpl.java

```diff
@@ -191,171 +214,219 @@ public SummariesType readLogSummaries(Integer folderId, String searchExpression)
-            log.setGroupLogs(groupLogs);
+	
+	
+	if (log.getStorage() == null) {
+	    // put extensions in constants
+	    String file_name = log.getFilePath() + "_" + log.getName() + ".xes.gz";
+	    String new_file_name = log.getFilePath() + "_" + newName + ".xes.gz";
+	    Storage storage = new Storage();
+	    storage.setStoragePath(config.getStoragePath());
+	    storage.setKey(new_file_name);
+	    storage.setPrefix("log");
+	    log.setStorage(storageRepository.saveAndFlush(storage));
 
-        } else if (!publicGroupLogs.isEmpty() && !isPublic) {
-            groupLogs.removeAll(publicGroupLogs);
-            log.setGroupLogs(groupLogs);
-        }
+	    StorageClient currentStorage = storageFactory.getStorageClient("FILE"+StorageType.STORAGE_PATH_SEPARATOR+ config.getLogsDir());
+	    StorageClient newStorage = storageFactory.getStorageClient(config.getStoragePath());
 
-        logRepo.saveAndFlush(log);
-    }
+	    OutputStream outputStream;
+
+	    outputStream = newStorage.getOutputStream("log", new_file_name);
+	    InputStream inputStream = currentStorage.getInputStream(null, file_name);
+	    logFileService.copyFile(inputStream, outputStream);
+	    
+	}
+	log.setName(newName);
 
-    private void updateLogName(Log log, String newName) {
-        String file_name = log.getFilePath() + "_" + log.getName() + ".xes.gz";
-        File file = new File(logsDir, file_name);
-        String new_file_name = log.getFilePath() + "_" + newName + ".xes.gz";
-        file.renameTo(new File(logsDir, new_file_name));
-        log.setName(newName);
     }
 
     @Override
     public boolean isPublicLog(Integer logId) {
-        return !filterPublicGroupLogs(logRepo.findUniqueByID(logId).getGroupLogs()).isEmpty();
+	return !filterPublicGroupLogs(logRepo.findUniqueByID(logId).getGroupLogs()).isEmpty();
     }
 
     private Set<GroupLog> filterPublicGroupLogs(Set<GroupLog> groupLogs) {
-        Group publicGroup = groupRepo.findPublicGroup();
-        if (publicGroup == null) {
-            LOGGER.warn("No public group present in repository");
-            return Collections.emptySet();
-        }
-
-        Set<GroupLog> publicGroupLogs = new HashSet<>(); /* groupLogs
-                .stream()
-                .filter(groupLog -> publicGroup.equals(groupLog.getGroup()))
-                .collect(Collectors.toSet());*/
-        for (GroupLog groupLog : groupLogs) {
-            if (publicGroup.equals(groupLog.getGroup())) {
-                publicGroupLogs.add(groupLog);
-            }
-        }
-
-        return publicGroupLogs;
+	Group publicGroup = groupRepo.findPublicGroup();
+	if (publicGroup == null) {
+	    LOGGER.warn("No public group present in repository");
+	    return Collections.emptySet();
+	}
+
+	Set<GroupLog> publicGroupLogs = new HashSet<>(); /*
+							  * groupLogs .stream() .filter(groupLog ->
+							  * publicGroup.equals(groupLog.getGroup())) .collect(Collectors.toSet());
+							  */
```

Comment:
I can see that we want to retain this stream-based alternative implementation for the glorious day when we ditch Virgo and can use lambda expressions, but the way it's been reformatted is bizarre to me.

Reply:
Will delete this

Reply:
No, don't delete it.  We're holding on to it so that when we ditch Virgo we can replace the Java 7 loop with newfangled streaming code.